File: test/language/parameterized_types/python/ExternParamTest.py
import zserio

import ParameterizedTypes

class ExternParamTest(ParameterizedTypes.TestCase):
    def testWriteRead(self):
        externData = zserio.BitBuffer(bytearray([0xCA, 0xFE]), 15)
        externParam = self.api.ExternParam(externData, self.api.Parameterized(externData, 13))

        bitBuffer = zserio.serialize(externParam)
        readExternParam = zserio.deserialize(self.api.ExternParam, bitBuffer)
        self.assertEqual(externParam, readExternParam)

File: test/extensions/language/parameterized_types/java/parameterized_types/extern_param/ExternParamTest.java
package parameterized_types.extern_param;

import static org.junit.jupiter.api.Assertions.*;

import java.io.IOException;

import org.junit.jupiter.api.Test;

import zserio.runtime.io.BitBuffer;
import zserio.runtime.io.SerializeUtil;

public class ExternParamTest
{
    @Test
    public void writeRead() throws IOException
    {
        final BitBuffer externData = new BitBuffer(new byte[] {(byte)0xCA, (byte)0xFE}, 15);
        final ExternParam externParam = new ExternParam(externData, new Parameterized(externData, 13));

        assertTrue(externParam.getExternField() == externParam.getParameterizedField().getParam());

        final BitBuffer bitBuffer = SerializeUtil.serialize(externParam);
        final ExternParam readExternParam = SerializeUtil.deserialize(ExternParam.class, bitBuffer);
        assertEquals(externParam, readExternParam);
    }
}
